### PR TITLE
Account for spark.executor.extraClassPath

### LINKED ISSUE #234: Respect spark.executor.extraClassPath
Possibly related to https://github.com/apache-spark-on-k8s/spark/issues/222#issuecomment-292270390

Mentioned as a TODO at https://github.com/apache-spark-on-k8s/spark/blob/branch-2.1-kubernetes/resource-managers/kubernetes/docker-minimal-bundle/src/main/docker/executor/Dockerfile#L39

Not entirely sure what this requires at this point wrt classpath ordering (the YARN cluster manager respects a `spark.driver.userClassPathFirst` setting)

cc @mccheah

### PR DESCRIPTION
**Pull request by mccheah**
Closes #234.

### CODE REVIEW

File: resource-managers/kubernetes/docker-minimal-bundle/src/main/docker/executor/Dockerfile [hunk 1]
```diff
@@ -37,4 +37,6 @@ ENV SPARK_HOME /opt/spark
 WORKDIR /opt/spark
 
 # TODO support spark.executor.extraClassPath
-CMD exec ${JAVA_HOME}/bin/java -Dspark.executor.port=$SPARK_EXECUTOR_PORT -Xms$SPARK_EXECUTOR_MEMORY -Xmx$SPARK_EXECUTOR_MEMORY -cp ${SPARK_HOME}/jars/\* org.apache.spark.executor.CoarseGrainedExecutorBackend --driver-url $SPARK_DRIVER_URL --executor-id $SPARK_EXECUTOR_ID --cores $SPARK_EXECUTOR_CORES --app-id $SPARK_APPLICATION_ID --hostname $SPARK_EXECUTOR_POD_IP
+CMD SPARK_EXECUTOR_CLASSPATH="${SPARK_HOME}/jars/*" && \
+    if ! [ -z ${SPARK_EXECUTOR_EXTRA_CLASSPATH+x} ]; then SPARK_EXECUTOR_CLASSPATH="$SPARK_EXECUTOR_EXTRA_CLASSPATH:$SPARK_EXECUTOR_CLASSPATH"; fi && \
```
Comment (foxish) [inline resource-managers/kubernetes/docker-minimal-bundle/src/main/docker/executor/Dockerfile]: Seems like `${SPARK_EXECUTOR_EXTRA_CLASSPATH+x}` isn't quite right

Comment (mccheah) [reply resource-managers/kubernetes/docker-minimal-bundle/src/main/docker/executor/Dockerfile]: We also do this in the driver docker file, to check the existence of an environment variable before using it.

File: resource-managers/kubernetes/docker-minimal-bundle/src/main/docker/executor/Dockerfile [hunk 2]
```diff
@@ -37,4 +37,6 @@ ENV SPARK_HOME /opt/spark
 WORKDIR /opt/spark
 
 # TODO support spark.executor.extraClassPath
```
Comment (ash211) [inline resource-managers/kubernetes/docker-minimal-bundle/src/main/docker/executor/Dockerfile]: can get rid of this TODO now?

**Review by ash211 — COMMENTED** [review]
Going to test with my internal app to check this works, but looks reasonable

**Comment by ash211** [comment]
Doesn't seem to work -- I think we need to set `--user-class-path` as a parameter to the `CoarseGrainedExecutorBackend` in the executor image.  Otherwise those jars aren't added to the classloader used for tasks that's created in `Executor.scala:createClassLoader()`

**Comment by mccheah** [comment]
A more comprehensive solution is required here.

**Pull request closed without merging**